src/error.rs

```diff
@@ -49,29 +52,77 @@ impl std::error::Error for HexToBytesError {
     fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
         use HexToBytesError::*;
 
-        match self {
-            InvalidChar(_) | OddLengthString(_) => None,
+        match *self {
+            InvalidChar(ref e) => Some(e),
+            OddLengthString(ref e) => Some(e),
         }
     }
 }
 
-/// Hex decoding error.
+impl From<InvalidCharError> for HexToBytesError {
+    #[inline]
+    fn from(e: InvalidCharError) -> Self { Self::InvalidChar(e) }
+}
+
+impl From<OddLengthStringError> for HexToBytesError {
+    #[inline]
+    fn from(e: OddLengthStringError) -> Self { Self::OddLengthString(e) }
+}
+
+/// Invalid hex character.
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[non_exhaustive]
```

Comment:
This is not needed when at least one field is *not*  `pub` but I could understand if someone wants to avoid us forgetting to add it in the future.

Reply:
I'm inclined to put it on every struct regardless because it makes scanning the error types quicker. However, this code is plain old sloppy, I've been not adding `Copy` when using `non_exhaustive`.

Reply:
I think naming error type `*Error` is much better. There are other types that need to be `non_exhaustive` and are not error types.

Reply:
I don't follow your comment, all errors are suffixed with `Error`. I'll remove the `non_exhaustive` from all struct errors that exist solely to hide internals. On further thought today it is strange to have unneeded `non_exhaustive` and there will always be private fields since that is the whole point of these types.

Reply:
Yes, they are and should continue to be. That's my point `rg 'struct [A-Za-z0-9]*Error' is better than `rg non_exhaustive`